src/structs/API.ts

```diff
@@ -0,0 +1,314 @@
+import fetch, { Headers } from 'node-fetch'
+import APIError from '../utils/APIError'
+import ClientUtils, { ClientType } from '../utils/ClientUtils'
+import qs from 'querystring'
+import '../typings'
+import { EventEmitter } from 'events'
+
+interface APIOptions {
+  /**
+   * Top.GG Token
+   */
+  token: any
+  /**
+   * Interval at which to post in milliseconds
+   * @default 1800000 30 minutes
+   */
+  interval?: number
+  /**
+   * Whether or not to autopost
+   * @default true
+   */
+  autoPost?: boolean
+  /**
+   * Whether or not in testing phase (devs only)
+   */
+  debug?: boolean
+}
+
+/**
+ * Top.GG API Client for Posting stats or Fetching data
+ * @link https://top.gg/api/docs
+ * @example
+ * ```js
+ * const client = Discord.Client() // Your discord.js or Eris client
+ *
+ * const TopGG = require('@top-gg/sdk')
+ *
+ * const api = new TopGG.API('Your top.gg token', client)
+ *
+ * api.on('posted', () => {
+ *   console.log('Server count posted!')
+ * })
+ * ```
+ */
+export default class API extends EventEmitter {
+  private options: APIOptions
+  private client: any
+  private clientType: ClientType
+  /**
+   * Create Top.GG API instance
+   * @param token Token or options
+   * @param client Supported library client 
+   */
+  constructor (token?: string|APIOptions, client?: any) {
+    super()
+    const defaultOptions = {
+      interval: 1800000,
+      autoPost: true,
+      debug: false
+    }
+    if (!token || typeof token === 'string') {
+      this.options = {
+        token,
+        ...defaultOptions
+      }
+    } else {
+      this.options = {
+        ...defaultOptions,
+        ...token
+      }
+    }
+
+    if (this.options.interval < 900000) throw new Error('Interval can not be lower than 900000 (15 minutes)')
+
+    this.client = client 
+    this.clientType = ClientUtils.figureClient(client)
+
+    if (this.options.debug) console.debug(`[DEBUG] Client Type: ${this.clientType}`)
+
+    if (this.options.autoPost && client) ClientUtils.setupAutoPoster(client, this.clientType, () => {
+      this.postStats()
+
+      setInterval(() => {
+        if (ClientUtils.isStarted(client, this.clientType)) this.postStats()
+      }, this.options.interval)
+    })
+  }
+
+  private async _request (method: string, path: string, body?: object): Promise<any> {
+    const headers = new Headers()
+    if (this.options.token) headers.set('Authorization', this.options.token)
+    if (method !== 'GET') headers.set('Content-Type', 'application/json')
+
+    let url = `https://top.gg/api/${path}`
```

Comment:
make this a const pls

Reply:
![image](https://user-images.githubusercontent.com/7953076/100909714-0f86cd80-349b-11eb-8c18-2bcbadc0ca51.png)
cant
+=
![image](https://user-images.githubusercontent.com/7953076/100909787-1e6d8000-349b-11eb-8067-6fc2da56ef31.png)